src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -1920,13 +1872,17 @@ int D4NFilterObject::D4NFilterDeleteOp::delete_obj(const DoutPrefixProvider* dpp
       }
       std::vector<std::string> responses;
       ret = blockDir->exec(dpp, responses, y);
-      if (ret < 0) {
+      if (ret < 0 && ret != -ENOENT) {
         ldpp_dout(dpp, 0) << "D4NFilterObject::" << __func__ << "(): Failed to execute exec in block directory: " << "ret= " << ret << dendl;
         return ret;
       }
       if (objDirty) {
-        if ((ret = delete_from_cache_and_policy(dpp, head_oid_in_cache, policy_prefix, y)) < 0) {
-          return ret;
+	if (!source->driver->get_policy_driver()->get_cache_policy()->invalidate_dirty_object(dpp, policy_prefix)) {
```

Comment:
Please refer to the comment in the latest delete_obj code and integrate invalidate_object